### ISSUE TITLE
[BUG]Mismatch shape and results when negative axis

### ISSUE THREAD

**Issue by YoshieraHuang** [mars-project/mars#1492]
## Describe the bug
When i use tensor.linalg.norm, I found some mismatching between derived tensor shape and result from execution, when i assigned a negative axis. 

## To Reproduce
To help us reproducing this bug, please provide information below:
1. Your Python version
3.7.7
2. The version of Mars you use
0.4.2
3. Versions of crucial packages, such as numpy, scipy and protobuf
numpy: 1.18.5
scipy: 1.5.0
4. Full stack of the error.
5. Minimized code to reproduce the error.
```
In []: mt.linalg.norm(((1,1,1)), axis = -1).shape
Out[]: (3,)
In []: mt.linalg.norm(((1,1,1)), axis = -1).execute()
Out[]: 1.7320508075688772
In []: mt.linalg.norm(((1,1,1),(1,-1,1)), axis = -1).shape                         
Out[]: (2, 3)
In []: mt.linalg.norm(((1,1,1),(1,-1,1)), axis = -1).execute()                     
Out[]: array([1.73205081, 1.73205081])
```

## Expected behavior
the shape should be (1,) and (2,)

**Comment by qinxuye** [comment]
It seems that we forgot to validate axis for `mt.linalg.norm`.

Effected code is:

https://github.com/mars-project/mars/blob/50feb684c843fa21da6b7470e305e6acc6f6917c/mars/tensor/linalg/norm.py#L305-L309

There is already an util to the validation work:

https://github.com/mars-project/mars/blob/50feb684c843fa21da6b7470e305e6acc6f6917c/mars/tensor/utils.py#L103-L107

Are you interested to fix the bug? I really encourage you to do this, if so, we would be very grateful about that.

BTW, don't forget to add unittests here:

https://github.com/mars-project/mars/blob/50feb684c843fa21da6b7470e305e6acc6f6917c/mars/tensor/linalg/tests/test_linalg_execute.py#L790

**Comment by YoshieraHuang** [comment]
I try to fix this bug but i cannot even run the script under the repository due to import errors. I am tired to fix the import errors. So i hpoe somebody can fix this.

**Comment by hekaisheng** [comment]
> I try to fix this bug but i cannot even run the script under the repository due to import errors. I am tired to fix the import errors. So i hpoe somebody can fix this.

https://docs.pymars.org/en/latest/development/contributing.html#contributing-to-mars this section may help you.

**Comment by qinxuye** [comment]
To simplify the process, assume the path you cloned Mars is: `/to/path/mars`. Then 

```bash
cd /to/path/mars
```

Make sure the libraries required by development installed.

```bash
pip install -r requirements-dev.txt
```

Then build protobuf files and compile the code.

```bash
python setup.py build_proto
python setup.py build_ext -i
```

Now you should be able to import mars normally. Just enjoy it.

**Comment by YoshieraHuang** [comment]
I fixed this bug at commit #1499

**Comment by qinxuye** [comment]
> I fixed this bug at commit #1499

Thank you so much, it looks great to me, I have some tiny comments, you can resolve it a little bit. Thank you again for your excellent work.
